src/app.py

```diff
@@ -0,0 +1,94 @@
+import json
+import logging
+import os
+import sys
+import urllib
+
+from flask import Flask, jsonify, render_template, request
+
+from index import load_index, query_engine
+
+app = Flask("goat_nlp")
+
+handler = logging.StreamHandler(sys.stdout)
+handler.setFormatter(
+    logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+)
+app.logger.addHandler(handler)
+app.logger.setLevel(logging.INFO)
+
+
+def construct_url(json_output):
+    base_url = "https://goat.genomehubs.org/"
+    endpoint = "search?"
+    suffix = (
+        "&result=taxon&summaryValues=count&taxonomy=ncbi&offset=0"
+        + "&fields=assembly_level%2Cassembly_span%2Cgenome_size%2C"
+        + "chromosome_number%2Chaploid_number&names=common_name&ranks="
+        + "&includeEstimates=false&size=100"
+    )
+
+    if json_output["intent"] == "count":
+        endpoint = "count?"
+    elif json_output["intent"] == "record":
+        endpoint = "record?"
+
+    params = []
+
+    if "taxon" in json_output:
+        params.append(f"tax_tree(* {json_output['taxon']})")

```

Comment:
fine for now - for next iteration use tax_name search to replace user tax name with taxon_id